open-ce/build_tree.py

```diff
@@ -345,8 +347,44 @@ def _clone_repo(self, git_url, repo_dir, env_config_data, git_tag_from_config):
 
         print("Clone cmd: ", clone_cmd)
         clone_result = os.system(clone_cmd)
-        if clone_result != 0:
-            raise OpenCEError(Error.CLONE_REPO, git_url)
+        cur_dir = os.getcwd()
+        clone_successful = clone_result == 0
+        if not clone_successful:
+            if not git_tag is None:
```

Comment:
Is there a better way than catching a failed clone?
Maybe we could determine if its a commit hash ahead of time above, then make sure we issue the correct `clone` or `clone` + `checkout` commands?

Reply:
I searched for git documentation/references where this can be checked. But unfortunately, without cloning the repo, we can't really get to know if the specified value under `git tag` is a branch/tag name or a commit hash. Also, branch name can have as many characters as that of a commit hash (For e.g. we can create branch name with `499b5b28e0a349b9c351881224e34cfaacc463fd` which is actually a commit hash). 
One way is we add another field `commit_hash` in our environment file for it as an alternative to `git_tag`.  So, if `commit_hash` is specified, we can do clone + checkout.

Reply:
Couldn't we always do this step, and just skip using the `-b` option all together?

Reply:
Yes, we can do.

Reply:
ok, this is better.